core/src/main/kotlin/no/strooped/view/screen/TaskScreen.kt

```diff
@@ -70,21 +66,25 @@ label.getStyle().background = new Image(new Texture(labelColor)).getDrawable();*
             }
         }
         var colorClicked = false
-        for (i in colorOptions.indices) {
-            colorOptions[i]!!.onClick {
+        colorOptions.forEach { button ->
+            button!!.onClick {
                 if (!colorClicked) {
-                    for (j in colorOptions.indices) {
-                        if (i != j) {
-                            colorOptions[j]!!.changeColor("#737373 ") // gray 87
-                        }
-                    }
-                    colorClicked = true
+                    focusOnButton(button)
                 }
+                colorClicked = true
             }
         }
         Gdx.input.inputProcessor = ui
     }
 
+    private fun focusOnButton(button: Button) {
+        val (otherColors, rest) = colorOptions.partition { it != null && it != button }
```

Comment:
I don't think partition is the correct function to use here, since we are really just treating it as a filter.
Instead you can chain filter functions.

```kotlin
val otherColors = colorOptions
            .filter { it != button }
            .filterNotNull()

        otherColors.forEach {
            it.changeColor("#737373")
        }
```

Reply:
Done